EventGenerator/src/GenFilter_module.cc

```diff
@@ -75,42 +109,70 @@ namespace mu2e {
 
   bool GenFilter::filter(art::Event& event) {
     if(isNull_) return true;
-    bool passed = false;
+    if(!(maxr_cut_ || energy_cut_ || momentum_cut_ || pT_cut_)) return true;
     auto sim = event.getValidHandle<StageParticleCollection>(_SimToken);
     _SimCol = sim.product();
+    bool pass_maxr_cut = !maxr_cut_;
+    bool pass_energy_cut = !energy_cut_;
+    bool pass_momentum_cut = !momentum_cut_;
+    bool pass_pT_cut = !momentum_cut_;
     for(const auto& aParticle : *_SimCol){
-    //  make momentum and position vectors
+      //  make momentum and position vectors
       GeomHandle<DetectorSystem> det;
       ROOT::Math::XYZVectorF pos = XYZVectorF(det->toDetector(aParticle.position()));
       ROOT::Math::XYZTVector pos0(pos.x(), pos.y(), pos.z(), aParticle.time());
       ROOT::Math::PxPyPzMVector mom0(aParticle.momentum().x(), aParticle.momentum().y(), aParticle.momentum().z(), aParticle.momentum().t());
+      _energy = aParticle.momentum().e();
+      _mom = aParticle.momentum().rho();
+      _momT = sqrt(mom0.x()*mom0.x() + mom0.y()*mom0.y());
 
-      // extract charge
-      static GlobalConstantsHandle<ParticleDataList> pdt;
-      auto charge = pdt->particle(aParticle.pdgId()).charge();
+      if(maxr_cut_) { // cut on maximum R in the solenoid
+        // extract charge
+        static GlobalConstantsHandle<ParticleDataList> pdt;
+        auto charge = pdt->particle(aParticle.pdgId()).charge();
 
-      // extact field
-      GeomHandle<BFieldManager> bfmgr;
-      mu2e::GeomHandle<mu2e::Tracker> tracker;
-      auto tracker_origin = det->toMu2e(tracker->origin());
-      //XYZVectorF pos3Vec = XYZVectorF(aParticle.position().x(),aParticle.position().y(),aParticle.position().z());
-      ROOT::Math::XYZVector bnom(bfmgr->getBField(tracker_origin));
+        // check if the particle is charged, else ignore this cut
+        if(fabs(charge) > 1.e-5) {
+          // extact field
+          GeomHandle<BFieldManager> bfmgr;
+          mu2e::GeomHandle<mu2e::Tracker> tracker;
+          auto tracker_origin = det->toMu2e(tracker->origin());
+          //XYZVectorF pos3Vec = XYZVectorF(aParticle.position().x(),aParticle.position().y(),aParticle.position().z());
+          ROOT::Math::XYZVector bnom(bfmgr->getBField(tracker_origin));
```

Comment:
`auto bnom = bfmgr->`  returns a `CLHEP::Hep3Vector` due to `tracker_origin` being this type, while `LoopHelix` requires a `XYZVector` reference so this can't be converted when passing (at least this is the complaint I see in compiling).